### ISSUE TITLE
Parasoll battery id does not contain device name

### ISSUE THREAD

**Issue by rhoriguchi** [sanjoyg/dirigera_platform#99]
I have a Parasol sensor connected to my hub named `Bedroom Window 1`. This now gets picked up by the component. The issue is that it's called `sensor.battery_percentage` and does not have a prefix with the name like the contact sensor `binary_sensor.bedroom_window_1`.

Side note, with the Home Kit component it's called `binary_sensor.bedroom_window_1_contact` and `sensor.bedroom_window_1_battery`.

---

Edit: Updated the example so it matches dumped log lines.

**Comment by sanjoyg** [comment]
Can dump json for me

**Comment by rhoriguchi** [comment]
I've replaced all IDs with random strings in the same format or UUID.

```
DeviceEntry(area_id='Bedroom', config_entries={'KDLPZCF38MIV9RLEZBRWUMJ9WN'}, configuration_url=None, connections=set(), created_at=datetime.datetime(2024, 10, 23, 0, 57, 3, 159533, tzinfo=datetime.timezone.utc), disabled_by=None, entry_type=None, hw_version=None, id='k3cav25ses6wt5qnumlzad4nhbv1eva5', identifiers={('dirigera_platform', '00cbb95f-0091-4b84-baf6-8fde9b56b3bd_1')}, labels=set(), manufacturer='IKEA of Sweden', model='PARASOLL Door/Window Sensor', model_id=None, modified_at=datetime.datetime(2024, 10, 23, 0, 57, 3, 159579, tzinfo=datetime.timezone.utc), name_by_user=None, name='Bedroom Window 1', primary_config_entry='KDLPZCF38MIV9RLEZBRWUMJ9WN', serial_number=None, suggested_area='Bedroom', sw_version='1.0.19', via_device_id=None, is_new=False)

id='00cbb95f-0091-4b84-baf6-8fde9b56b3bd_1' relation_id=None type='sensor' device_type='openCloseSensor' created_at=datetime.datetime(2024, 4, 3, 20, 7, 20, tzinfo=TzInfo(UTC)) is_reachable=True last_seen=datetime.datetime(2024, 10, 23, 10, 36, 2, tzinfo=TzInfo(UTC)) attributes=OpenCloseSensorAttributes(custom_name='Bedroom Window 1', model='PARASOLL Door/Window Sensor', manufacturer='IKEA of Sweden', firmware_version='1.0.19', hardware_version='1', serial_number='7APMLP30I6XVUO0U', product_code='E2013', ota_status=None, ota_state=None, ota_progress=None, ota_policy=None, ota_schedule_start=None, ota_schedule_end=None, is_open=True, battery_percentage=100) capabilities=Capabilities(can_send=[], can_receive=['customName']) room=Room(id='00a7292d-9ae8-4af6-a337-c0d69e9d3b37', name='Bedroom', color='ikea_white_no_20', icon='rooms_arm_chair') device_set=[] remote_links=[] is_hidden=False dirigera_client=<custom_components.dirigera_platform.dirigera_lib_patch.HubX object at 0x7fb8ce638290>

id='00cbb95f-0091-4b84-baf6-8fde9b56b3bd_1' relation_id=None type='sensor' device_type='openCloseSensor' created_at=datetime.datetime(2024, 4, 3, 20, 7, 20, tzinfo=TzInfo(UTC)) is_reachable=True last_seen=datetime.datetime(2024, 10, 23, 10, 36, 2, tzinfo=TzInfo(UTC)) attributes=OpenCloseSensorAttributes(custom_name='Bedroom Window 1', model='PARASOLL Door/Window Sensor', manufacturer='IKEA of Sweden', firmware_version='1.0.19', hardware_version='1', serial_number='7APMLP30I6XVUO0U', product_code='E2013', ota_status=None, ota_state=None, ota_progress=None, ota_policy=None, ota_schedule_start=None, ota_schedule_end=None, is_open=True, battery_percentage=100) capabilities=Capabilities(can_send=[], can_receive=['customName']) room=Room(id='00a7292d-9ae8-4af6-a337-c0d69e9d3b37', name='Bedroom', color='ikea_white_no_20', icon='rooms_arm_chair') device_set=[] remote_links=[] is_hidden=False dirigera_client=<custom_components.dirigera_platform.dirigera_lib_patch.HubX object at 0x7fb8ce638290>
id='00cbb95f-0091-4b84-baf6-8fde9b56b3bd_1' relation_id=None type='sensor' device_type='openCloseSensor' created_at=datetime.datetime(2024, 4, 3, 20, 7, 20, tzinfo=TzInfo(UTC)) is_reachable=True last_seen=datetime.datetime(2024, 10, 23, 10, 36, 2, tzinfo=TzInfo(UTC)) attributes=OpenCloseSensorAttributes(custom_name='Bedroom Window 1', model='PARASOLL Door/Window Sensor', manufacturer='IKEA of Sweden', firmware_version='1.0.19', hardware_version='1', serial_number='7APMLP30I6XVUO0U', product_code='E2013', ota_status=None, ota_state=None, ota_progress=None, ota_policy=None, ota_schedule_start=None, ota_schedule_end=None, is_open=False, battery_percentage=100) capabilities=Capabilities(can_send=[], can_receive=['customName']) room=Room(id='00a7292d-9ae8-4af6-a337-c0d69e9d3b37', name='Bedroom', color='ikea_white_no_20', icon='rooms_arm_chair') device_set=[] remote_links=[] is_hidden=False dirigera_client=<custom_components.dirigera_platform.dirigera_lib_patch.HubX object at 0x7fb8ce638290>
```

**Comment by sanjoyg** [comment]
Please use the instructions [here](https://github.com/sanjoyg/dirigera_platform/wiki/Calling-dump_data-to-dump-the-JSON) to dump the json and paste it in the issue

**Comment by rhoriguchi** [comment]
Here you go, replaced ids like on top and removed all other devices.

```
  [
    ...

    {
      'id': '00cbb95f-0091-4b84-baf6-8fde9b56b3bd_1',
      'type': 'sensor',
      'deviceType': 'openCloseSensor',
      'createdAt': '2024-04-03T20:07:20.000Z',
      'isReachable': True,
      'lastSeen': '2024-10-24T22:22:36.000Z',
      'customIcon': 'placement_door',
      'attributes': {
        'customName': 'Bedroom Window 1',
        'model': 'PARASOLL Door/Window Sensor',
        'manufacturer': 'IKEA of Sweden',
        'firmwareVersion': '1.0.19',
        'hardwareVersion': '1',
        'serialNumber': '7APMLP30I6XVUO0U',
        'productCode': 'E2013',
        'batteryPercentage': 100,
        'identifyStarted': '2000-01-01T00:00:00.000Z',
        'identifyPeriod': 0,
        'isOpen': True,
        'permittingJoin': False
      },
      'capabilities': {
        'canSend': [],
        'canReceive': [
          'customName'
        ]
      },
      'room': {
        'id': '00a7292d-9ae8-4af6-a337-c0d69e9d3b37',
        'name': 'Bedroom',
        'color': 'ikea_white_no_20',
        'icon': 'rooms_arm_chair'
      },
      'deviceSet': [],
      'remoteLinks': [],
      'isHidden': False
    }

    ...
  ]
```

**Comment by sanjoyg** [comment]
do dump the entire json. Update to new version it sanitizes the serial number and ids

**Comment by rhoriguchi** [comment]
Thank you for the update. I've attached the raw logs with no modification, the device is called `Reduit Parasoll`.

```
2024-10-26 04:50:41.484 INFO (SyncWorker_2) [custom_components.dirigera_platform] [{'id': '63ea675f-b7ef-41af-bb7b-76b72113041b_1', 'type': 'outlet', 'deviceType': 'outlet', 'createdAt': '2024-05-13T10:44:52.000Z', 'isReachable': True, 'lastSeen': '2024-10-26T00:43:12.000Z', 'attributes': {'customName': 'Kitchen standing lamp', 'model': 'TRADFRI control outlet', 'manufacturer': 'IKEA of Sweden', 'firmwareVersion': '2.3.089', 'hardwareVersion': '1', 'serialNumber': '3425B4FFFE94A6B2', 'productCode': 'E1603', 'isOn': False, 'startupOnOff': 'startPrevious', 'lightLevel': 100, 'startUpCurrentLevel': -1, 'identifyStarted': '2000-01-01T00:00:00.000Z', 'identifyPeriod': 0, 'permittingJoin': False, 'otaStatus': 'upToDate', 'otaState': 'readyToCheck', 'otaProgress': 0, 'otaPolicy': 'autoUpdate', 'otaScheduleStart': '00:00', 'otaScheduleEnd': '00:00'}, 'capabilities': {'canSend': [], 'canReceive': ['customName', 'isOn', 'lightLevel']}, 'room': {'id': '2f341004-7f6d-4dfd-b707-7a469c370105', 'name': 'Kitchen', 'color': 'ikea_lilac_no_3', 'icon': 'rooms_pot_with_lid'}, 'deviceSet': [], 'remoteLinks': [], 'isHidden': False}, {'id': 'b5016f8e-62eb-4816-adf7-ad76220afb08_1', 'type': 'outlet', 'deviceType': 'outlet', 'createdAt': '2024-07-13T20:48:44.000Z', 'isReachable': True, 'lastSeen': '2024-10-26T00:43:56.000Z', 'attributes': {'customName': 'Lego bonsai', 'model': 'TRADFRI control outlet', 'manufacturer': 'IKEA of Sweden', 'firmwareVersion': '2.3.089', 'hardwareVersion': '1', 'serialNumber': '3425B4FFFEC80731', 'productCode': 'E1603', 'isOn': True, 'startupOnOff': 'startPrevious', 'lightLevel': 100, 'startUpCurrentLevel': -1, 'identifyStarted': '2000-01-01T00:00:00.000Z', 'identifyPeriod': 0, 'permittingJoin': False, 'otaStatus': 'upToDate', 'otaState': 'readyToCheck', 'otaProgress': 0, 'otaPolicy': 'autoUpdate', 'otaScheduleStart': '00:00', 'otaScheduleEnd': '00:00'}, 'capabilities': {'canSend': [], 'canReceive': ['customName', 'isOn', 'lightLevel']}, 'room': {'id': '2579d2c3-803f-451d-8ac2-ba16cd24bc61', 'name': 'Lego', 'color': 'ikea_yellow_no_24', 'icon': 'rooms_play_area'}, 'deviceSet': [], 'remoteLinks': [], 'isHidden': False}, {'id': '754d3ba5-ab0a-4a87-b0ee-1fc945284b2a_1', 'type': 'sensor', 'deviceType': 'openCloseSensor', 'createdAt': '2024-04-03T20:07:20.000Z', 'isReachable': True, 'lastSeen': '2024-10-25T22:09:28.000Z', 'customIcon': 'placement_door', 'attributes': {'customName': 'Reduit Parasoll', 'model': 'PARASOLL Door/Window Sensor', 'manufacturer': 'IKEA of Sweden', 'firmwareVersion': '1.0.19', 'hardwareVersion': '1', 'serialNumber': '048727FFFE527BF4', 'productCode': 'E2013', 'batteryPercentage': 100, 'identifyStarted': '2000-01-01T00:00:00.000Z', 'identifyPeriod': 0, 'isOpen': True, 'permittingJoin': False}, 'capabilities': {'canSend': [], 'canReceive': ['customName']}, 'room': {'id': '73b27316-ca76-49d6-9a53-19221349de9b', 'name': 'Reduit', 'color': 'ikea_white_no_20', 'icon': 'rooms_arm_chair'}, 'deviceSet': [], 'remoteLinks': [], 'isHidden': False}, {'id': 'fa009194-91ca-41fb-adb2-730babe94809_1', 'type': 'light', 'deviceType': 'light', 'createdAt': '2024-05-23T22:47:28.000Z', 'isReachable': True, 'lastSeen': '2024-10-25T10:35:30.000Z', 'customIcon': 'products_driver', 'attributes': {'customName': 'Reduit closet lights', 'model': 'TRADFRI Driver 10W', 'manufacturer': 'IKEA of Sweden', 'firmwareVersion': '1.0.4', 'hardwareVersion': '1', 'serialNumber': '705464FFFE2BD0EC', 'productCode': 'ICPSHC2410EUIL2', 'isOn': False, 'startupOnOff': 'startPrevious', 'lightLevel': 100, 'startUpCurrentLevel': -1, 'identifyStarted': '2000-01-01T00:00:00.000Z', 'identifyPeriod': 0, 'permittingJoin': False, 'otaStatus': 'upToDate', 'otaState': 'readyToCheck', 'otaProgress': 0, 'otaPolicy': 'autoUpdate', 'otaScheduleStart': '00:00', 'otaScheduleEnd': '00:00'}, 'capabilities': {'canSend': [], 'canReceive': ['customName', 'isOn', 'lightLevel']}, 'room': {'id': '73b27316-ca76-49d6-9a53-19221349de9b', 'name': 'Reduit', 'color': 'ikea_white_no_20', 'icon': 'rooms_arm_chair'}, 'deviceSet': [], 'remoteLinks': [], 'isHidden': False, 'adaptiveProfile': {}}, {'id': '401e2fe2-8979-4ae3-96b5-5614dcf35e6e_1', 'type': 'outlet', 'deviceType': 'outlet', 'createdAt': '2024-04-14T13:31:30.000Z', 'isReachable': True, 'lastSeen': '2024-10-25T18:16:43.000Z', 'attributes': {'customName': 'Entrance sideboard lamp', 'model': 'TRADFRI control outlet', 'manufacturer': 'IKEA of Sweden', 'firmwareVersion': '2.3.089', 'hardwareVersion': '1', 'serialNumber': '6C5CB1FFFE59E4BB', 'productCode': 'E1603', 'isOn': False, 'startupOnOff': 'startPrevious', 'lightLevel': 100, 'startUpCurrentLevel': -1, 'identifyStarted': '2000-01-01T00:00:00.000Z', 'identifyPeriod': 0, 'permittingJoin': False, 'otaStatus': 'upToDate', 'otaState': 'readyToCheck', 'otaProgress': 0, 'otaPolicy': 'autoUpdate', 'otaScheduleStart': '00:00', 'otaScheduleEnd': '00:00'}, 'capabilities': {'canSend': [], 'canReceive': ['customName', 'isOn', 'lightLevel']}, 'room': {'id': '911a6fe3-89d8-44f5-93e4-b0d1e5386de5', 'name': 'Entrance', 'color': 'ikea_pink_no_8', 'icon': 'rooms_coat_rack'}, 'deviceSet': [], 'remoteLinks': [], 'isHidden': False}, {'id': '5fcfac01-e89d-4c34-b8a6-b0ff14958183_1', 'type': 'light', 'deviceType': 'light', 'createdAt': '2024-10-01T19:48:28.000Z', 'isReachable': True, 'lastSeen': '2024-10-25T17:48:53.000Z', 'customIcon': 'products_led_bulb', 'attributes': {'customName': 'Hallway lamp', 'model': 'TRADFRI bulb E14 CWS globe 806lm', 'manufacturer': 'IKEA of Sweden', 'firmwareVersion': '1.0.38', 'hardwareVersion': '1', 'serialNumber': '842712FFFE2B2BB4', 'productCode': 'LED2111G6', 'isOn': False, 'startupOnOff': 'startPrevious', 'lightLevel': 100, 'startUpCurrentLevel': -1, 'colorHue': 28.7677001953125, 'colorSaturation': 0.5708661417322834, 'colorTemperature': 2702, 'colorTemperatureMin': 4000, 'colorTemperatureMax': 2202, 'startupTemperature': -1, 'colorMode': 'color', 'identifyStarted': '2000-01-01T00:00:00.000Z', 'identifyPeriod': 0, 'permittingJoin': False, 'otaStatus': 'upToDate', 'otaState': 'readyToCheck', 'otaProgress': 0, 'otaPolicy': 'autoUpdate', 'otaScheduleStart': '00:00', 'otaScheduleEnd': '00:00'}, 'capabilities': {'canSend': [], 'canReceive': ['customName', 'isOn', 'lightLevel', 'colorTemperature', 'colorHue', 'colorSaturation']}, 'room': {'id': 'f5692a41-2fea-491a-a8f3-2abfeb0fde14', 'name': 'Hallway', 'color': 'ikea_yellow_no_31', 'icon': 'rooms_door'}, 'deviceSet': [], 'remoteLinks': [], 'isHidden': False, 'adaptiveProfile': {}}, {'id': 'f80cc620-f30c-4152-bfd6-49ee6403eb2b_1', 'type': 'light', 'deviceType': 'light', 'createdAt': '2024-04-07T19:23:33.000Z', 'isReachable': True, 'lastSeen': '2024-10-25T04:24:46.000Z', 'attributes': {'customName': 'Bedroom closet lights', 'model': 'TRADFRI Driver 30W', 'manufacturer': 'IKEA of Sweden', 'firmwareVersion': '2.3.086', 'hardwareVersion': '1', 'serialNumber': '5C0272FFFE8A74D6', 'productCode': 'ICPSHC2430EUIL1', 'isOn': True, 'startupOnOff': 'startPrevious', 'lightLevel': 100, 'startUpCurrentLevel': -1, 'identifyStarted': '2000-01-01T00:00:00.000Z', 'identifyPeriod': 0, 'permittingJoin': False, 'otaStatus': 'upToDate', 'otaState': 'readyToCheck', 'otaProgress': 0, 'otaPolicy': 'autoUpdate', 'otaScheduleStart': '00:00', 'otaScheduleEnd': '00:00'}, 'capabilities': {'canSend': [], 'canReceive': ['customName', 'isOn', 'lightLevel']}, 'room': {'id': '1f7abb3e-2b97-4436-9cc6-5d50b9584147', 'name': 'Bedroom', 'color': 'ikea_brown_no_41', 'icon': 'rooms_bed'}, 'deviceSet': [], 'remoteLinks': [], 'isHidden': False, 'adaptiveProfile': {}}, {'id': '1b41e42f-6768-4413-b475-aa64acd706c8_1', 'type': 'outlet', 'deviceType': 'outlet', 'createdAt': '2024-09-06T19:56:26.000Z', 'isReachable': True, 'lastSeen': '2024-10-26T01:14:50.000Z', 'attributes': {'customName': 'Entrance lamp', 'model': 'TRADFRI control outlet', 'manufacturer': 'IKEA of Sweden', 'firmwareVersion': '2.3.089', 'hardwareVersion': '1', 'serialNumber': '90395EFFFE0A5DEC', 'productCode': 'E1603', 'isOn': False, 'startupOnOff': 'startPrevious', 'lightLevel': 100, 'startUpCurrentLevel': -1, 'identifyStarted': '2000-01-01T00:00:00.000Z', 'identifyPeriod': 0, 'permittingJoin': False, 'otaStatus': 'upToDate', 'otaState': 'readyToCheck', 'otaProgress': 0, 'otaPolicy': 'autoUpdate', 'otaScheduleStart': '00:00', 'otaScheduleEnd': '00:00'}, 'capabilities': {'canSend': [], 'canReceive': ['customName', 'isOn', 'lightLevel']}, 'room': {'id': '911a6fe3-89d8-44f5-93e4-b0d1e5386de5', 'name': 'Entrance', 'color': 'ikea_pink_no_8', 'icon': 'rooms_coat_rack'}, 'deviceSet': [], 'remoteLinks': [], 'isHidden': False}, {'id': '3fb2ef61-d5a6-4c4c-862e-646014405e42_1', 'relationId': '3fb2ef61-d5a6-4c4c-862e-646014405e42', 'type': 'gateway', 'deviceType': 'gateway', 'createdAt': '2024-03-17T18:58:30.942Z', 'isReachable': True, 'lastSeen': '2024-10-26T02:50:00.363Z', 'attributes': {'customName': 'Home', 'model': 'DIRIGERA Hub for smart products', 'manufacturer': 'IKEA of Sweden', 'firmwareVersion': '2.660.5', 'hardwareVersion': 'P2.5', 'serialNumber': '3fb2ef61-d5a6-4c4c-862e-646014405e42', 'identifyStarted': '2000-01-01T00:00:00.000Z', 'identifyPeriod': 0, 'otaStatus': 'upToDate', 'otaState': 'readyToCheck', 'otaProgress': 100, 'otaPolicy': 'autoDownload', 'otaScheduleStart': '00:00', 'otaScheduleEnd': '00:00', 'permittingJoin': False, 'backendConnected': True, 'backendConnectionPersistent': True, 'backendOnboardingComplete': True, 'backendRegion': 'eu-central-1', 'backendCountryCode': 'CH', 'userConsents': [{'name': 'analytics', 'value': 'disabled'}, {'name': 'diagnostics', 'value': 'enabled'}], 'logLevel': 3, 'coredump': False, 'timezone': 'Europe/Zurich', 'nextSunSet': None, 'nextSunRise': None, 'homestate': 'home', 'countryCode': 'XZ', 'isOn': False}, 'capabilities': {'canSend': [], 'canReceive': ['customName', 'permittingJoin', 'userConsents', 'logLevel', 'time', 'timezone', 'countryCode', 'coordinates']}, 'deviceSet': [], 'remoteLinks': []}, {'id': '4bc556da-2e1b-4cff-bd1c-a26d5a97302b_1', 'type': 'light', 'deviceType': 'light', 'createdAt': '2024-04-04T12:47:15.000Z', 'isReachable': True, 'lastSeen': '2024-10-25T18:35:05.000Z', 'customIcon': 'lighting_desk_lamp', 'attributes': {'customName': 'Living room table lamp', 'model': 'TRADFRI bulb E14 CWS globe 806lm', 'manufacturer': 'IKEA of Sweden', 'firmwareVersion': '1.0.38', 'hardwareVersion': '1', 'serialNumber': '60EFABFFFE33F0B0', 'productCode': 'LED2111G6', 'isOn': False, 'startupOnOff': 'startPrevious', 'lightLevel': 100, 'startUpCurrentLevel': -1, 'colorHue': 28.7677001953125, 'colorSaturation': 0.5708661417322834, 'colorTemperature': 2702, 'colorTemperatureMin': 4000, 'colorTemperatureMax': 2202, 'startupTemperature': -1, 'colorMode': 'color', 'identifyStarted': '2000-01-01T00:00:00.000Z', 'identifyPeriod': 0, 'permittingJoin': False, 'otaStatus': 'upToDate', 'otaState': 'readyToCheck', 'otaProgress': 0, 'otaPolicy': 'autoUpdate', 'otaScheduleStart': '00:00', 'otaScheduleEnd': '00:00'}, 'capabilities': {'canSend': [], 'canReceive': ['customName', 'isOn', 'lightLevel', 'colorTemperature', 'colorHue', 'colorSaturation']}, 'room': {'id': '8af56c5c-d5d4-444b-81d6-f4d6248ebb8b', 'name': 'Living room', 'color': 'ikea_blue_no_63', 'icon': 'rooms_arm_chair'}, 'deviceSet': [], 'remoteLinks': [], 'isHidden': False, 'adaptiveProfile': {}}, {'id': '9bf1e850-41f0-4028-967a-a8bcb20d653d_1', 'type': 'outlet', 'deviceType': 'outlet', 'createdAt': '2024-03-22T21:02:58.000Z', 'isReachable': True, 'lastSeen': '2024-10-26T00:41:34.000Z', 'attributes': {'customName': 'Living room standing lamp', 'model': 'TRADFRI control outlet', 'manufacturer': 'IKEA of Sweden', 'firmwareVersion': '2.3.089', 'hardwareVersion': '1', 'serialNumber': 'BC026EFFFE600061', 'productCode': 'E1603', 'isOn': False, 'startupOnOff': 'startPrevious', 'lightLevel': 100, 'startUpCurrentLevel': -1, 'identifyStarted': '2000-01-01T00:00:00.000Z', 'identifyPeriod': 0, 'permittingJoin': False, 'otaStatus': 'upToDate', 'otaState': 'readyToCheck', 'otaProgress': 0, 'otaPolicy': 'autoUpdate', 'otaScheduleStart': '00:00', 'otaScheduleEnd': '00:00'}, 'capabilities': {'canSend': [], 'canReceive': ['customName', 'isOn', 'lightLevel']}, 'room': {'id': '8af56c5c-d5d4-444b-81d6-f4d6248ebb8b', 'name': 'Living room', 'color': 'ikea_blue_no_63', 'icon': 'rooms_arm_chair'}, 'deviceSet': [], 'remoteLinks': [], 'isHidden': False}]
```

**Comment by sanjoyg** [comment]
Do you see any errors in the log? When I tried your JSON I do get the device to report the status (open/close) and the battery level

<img width="666" alt="Screenshot 2024-10-27 at 03 22 28" src="https://github.com/user-attachments/assets/d4b1d44f-bbdb-43bc-a7ab-7568fa21fad4">

**Comment by rhoriguchi** [comment]
It does work with no issue. The problem is the naming. It's called `sensor.battery_percentag`. It needs some name prefixing because if you have more than one, the second one will be called `sensor.battery_percentage_1`.

**Comment by sanjoyg** [comment]
Aah… ok

**Comment by sanjoyg** [comment]
ok fixed now in the latest release please try and let know

**Comment by sanjoyg** [comment]
re-open the issue if its not fixed

**Comment by rhoriguchi** [comment]
Finally could test it. The change does what it should. 

To test I've readded the integration. What I've noticed is that it changed the IDs of all other devices for example `switch.entrance_lamp` to `switch.entrance_lamp_entrance_lamp`. I would propose to add the prefix in c85b58262006bbd56386d3a3a655d345dfaef011 only if it is a `Parasoll` device or check if the value from `self._device.name != self._name`.
